internal/backend/sessions/sessionworkflows/callbacks_signals.go

```diff
@@ -93,10 +98,16 @@ func (w *sessionWorkflow) nextSignal(wctx workflow.Context) func(context.Context
 			return nil, wctx.Err()
 		}
 
-		if signal == nil {
+		if signal == nil || signal.Name == "" {
```

Comment:
What does it mean when a signal is `nil`? What does it mwan when it's not `nil` but has an empty name?